processor/src/test/java/org/mapstruct/ap/test/bugs/_1005/Issue1005Test.java

```diff
@@ -0,0 +1,103 @@
+    Order.class,
+    OrderDto.class
+})
+public class Issue1005Test {
+
+    @WithClasses(Issue1005ErroneousAbstractResultTypeMapper.class)
+    @Test
+    @ExpectedCompilationOutcome(value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = Issue1005ErroneousAbstractResultTypeMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 30,
+                messageRegExp = ".*\\.AbstractEntity is an abstract class that cannot be used to create an instance " +
+                    "for: .*\\.HasKey.")
+        })
+    public void shouldFailDueToAbstractResultType() throws Exception {
+    }
+
+    @WithClasses(Issue1005ErroneousAbstractReturnTypeMapper.class)
+    @Test
+    @ExpectedCompilationOutcome(value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = Issue1005ErroneousAbstractReturnTypeMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 29,
+                messageRegExp = "The return type .*\\.AbstractEntity is abstract. Provide non abstract result type or" +
+                    " a factory method.")
+        })
+    public void shouldFailDueToAbstractReturnType() throws Exception {
+    }
+
+    @WithClasses(Issue1005ErroneousInterfaceResultTypeMapper.class)
+    @Test
+    @ExpectedCompilationOutcome(value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = Issue1005ErroneousInterfaceResultTypeMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 30,
+                messageRegExp = ".*\\.HasPrimaryKey is an interface and cannot be used to create an instance for: " +
+                    ".*\\.HasKey."),
+            @Diagnostic(type = Issue1005ErroneousInterfaceResultTypeMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 31,
+                messageRegExp = "No implementation type is registered for return type .*\\.HasKey.")
+        })
+    public void shouldFailDueToInterfaceResultType() throws Exception {
+    }
+
+    @WithClasses(Issue1005ErroneousInterfaceReturnTypeMapper.class)
+    @Test
+    @ExpectedCompilationOutcome(value = CompilationResult.FAILED,
+        diagnostics = {
+            @Diagnostic(type = Issue1005ErroneousInterfaceReturnTypeMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 29,
+                messageRegExp = "The return type .*\\.HasKey is interface. Provide non abstract/interface result type" +
+                    " or a factory method."),
+            @Diagnostic(type = Issue1005ErroneousInterfaceReturnTypeMapper.class,
+                kind = javax.tools.Diagnostic.Kind.ERROR,
+                line = 29,
+                messageRegExp = "No implementation type is registered for return type .*\\.HasKey.")
+        })
+    public void shouldFailDueToInterfaceReturnType() throws Exception {
```

Comment:
Well we don't have to. However the second line comes from the [`MapperCreationProcessor#L538`](https://github.com/mapstruct/mapstruct/blob/6b07dda829788387abef3ff8574066553ad7b646/processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java#L538). If I return `null` in the `BeanMappingMethod` when an error occurs then there will be only one error. Is that OK?